staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -179,7 +179,8 @@ func getIntOrPercentValue(intOrStr *IntOrString) (int, bool, error) {
 		if err != nil {
 			return 0, false, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
 		}
-		return int(v), true, nil
+		return int(v), strings.Contains(intOrStr.StrVal, "%"), nil
```

Comment:
see   https://github.com/kubernetes/kubernetes/pull/89482/files#diff-31ae0ffb0640d8ca5eb91a4a72e8f152R177

it seems that we have replace the '%'

Reply:
```
case intstr.String:
		isPercent := false
		s := intOrStr.StrVal
		if strings.Contains(s, "%") {
			isPercent = true
			s = strings.Replace(intOrStr.StrVal, "%", "", -1)
		}
		v, err := strconv.Atoi(s)
		if err != nil {
			return 0, isPercent, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
		}
		return int(v), isPercent, nil
	}
```

Reply:
/cc @JoelSpeed

Reply:
https://github.com/kubernetes/kubernetes/pull/89482#discussion_r411065116 This suggestion seems good to me, though I would suggest swapping `strings.Contains` for `strings.HasSuffix` and `strings.Replace` for `strings.TrimSuffix` so that cases where a user has entered `1%0` throws an error rather than is interpreted as `10%`

Reply:
agree. could you think about it? it looks good.

Reply:
swapping strings.Contains for strings.HasSuffix and strings.Replace for strings.TrimSuffix . 
And add some testcase.

Reply:
Looks good to me. I'll make the changes asap

Reply:
commited the changes

Reply:
@tanjunchen @JoelSpeed did you agree with the implementation?

Reply:
Implementation looks good, I added a suggestion for an extra test case, you may also be inspired for some more test cases by https://github.com/openshift/machine-api-operator/blob/b9f8d296e5f713a8fe7c5b89013ab37a4ac69172/pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go#L2645-L2719

Reply:
Thanks, added some more test cases, commiting the changes now